process/transactionProcessor.go

```diff
@@ -509,11 +509,16 @@ func (tp *TransactionProcessor) computeTransactionStatus(tx *transaction.ApiTran
 }
 
 func checkIfFailedOnReturnMessage(allScrs []*transaction.ApiTransactionResult, tx *transaction.ApiTransactionResult) bool {
-	if len(tx.ReturnMessage) > 0 && isZeroValue(tx.Value) {
+	hasReturnMessageWithZeroValue := len(tx.ReturnMessage) > 0 && isZeroValue(tx.Value)
+	if hasReturnMessageWithZeroValue && !isRefundScr(tx.ReturnMessage) {
 		return true
 	}
 
 	for _, scr := range allScrs {
+		if isRefundScr(scr.ReturnMessage) {
+			continue
+		}
+
 		if len(scr.ReturnMessage) > 0 && isZeroValue(scr.Value) {
```

Comment:
in case of one single inner tx is failed here we return fail

I think we need to discuss if this is the best approach

Reply:
I would say that the relay tx status should be successful even if all inner txs were failed, if the relay tx itself did not fail.

Reply:
I agree with Adi here.